lib/eco/incparser/error_recovery.py

```diff
@@ -36,6 +36,10 @@ def recover(self, error_node):
         # Get the character offset of the top node on the stack
         error_offset = self.stack_offset(len(self.stack))
 
+        # XXX When searching for an isolation node and we discover an iso node on
```

Comment:
I meant an already isolated node (i.e. the algorithm chose an isolated tree that afterwards turns out to not be a big enough isolation so it needs to find a bigger isolation; in which case we need to unmark the previous isolation tree again). I'm not sure yet if this can actually happen. It happened once but only due to a bug so it disappeared after I fixed the bug. But I'm not 100% certain that there isn't a rare scenario that can cause this again.